ambry-store/src/main/java/com.github.ambry.store/StoreMessageReadSet.java

```diff
@@ -196,4 +237,13 @@ public StoreKey getKeyAt(int index) {
     }
     return readOptions.get(index).getMessageInfo().getStoreKey();
   }
+
+  @Override
+  public void doPrefetch(int index, long relativeOffset, long size) {
+    try {
+      readOptions.get(index).doPrefetch(relativeOffset, size);
+    } catch (Exception e) {
+      logger.error("Data Prefetch failed", e);
```

Comment:
so why is this an error and not a warning? Since the system proceeds without giving up, I think it should be a warning

Reply:
Checked the definition. Right, warn is appropriate here. 

Warn - Anything that can potentially cause application oddities, but for which I am automatically recovering. (Such as switching from a primary to backup server, retrying an operation, missing secondary data, etc.)

Error - Any error which is fatal to the operation, but not the service or application (can't open a required file, missing data, etc.). These errors will force user (administrator, or direct user) intervention. These are usually reserved (in my apps) for incorrect connection strings, missing services, etc.